packagedb/api.py

```diff
@@ -419,6 +421,99 @@ def filter_by_checksums(self, request, *args, **kwargs):
         return self.get_paginated_response(serialized_package_data)
 
 
+class PackageViewSet(PackagePublicViewSet):
+    @action(detail=True)
+    def reindex_package(self, request, *args, **kwargs):
+
+        """
+        Reindex this package instance
+        """
+        package = self.get_object()
+        package.reindex()
+        data = {
+            'status': f'{package.package_url} has been queued for reindexing'
+        }
+        return Response(data)
+
+
+class PackageUpdateSet(viewsets.ViewSet):
+
+    """
+    Take a list of `purls` (where each item is a dictionary containing PURL
+    and content_type).
+
+    If `uuid` is given then all purls will be added to package set if it exists
+    else a new set would be created and all the purls will be added to that new set.
+
+    **Note:** There is also a slight addition to the logic where a purl already exists in the database
+    and so there are no changes done to the purl entry it is passed as it is.
+
+    **Request example:**
+        {
+          "purls": [
+            {"purl": "pkg:npm/less@1.0.32", "content_type": 1}
+          ],
+          "uuid" : "b67ceb49-1538-481f-a572-431062f382gg"
+        }
+    """
+
+    def create(self, request):
+
+        res = []
+
+        serializer = UpdatePackagesSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response({'errors': serializer.errors}, status=400)
+
+        validated_data = serializer.validated_data
+        packages = validated_data.get('purls', [])
+        uuid = validated_data.get('uuid', None)
+        package_set = None
+
+        if uuid:
+            try:
+                package_set = PackageSet.objects.get(uuid=uuid)
+
+                if package_set:
+                    package_set = package_set
+
+            except:
+                message = {
+                    'status': f'No Package Set found for {uuid}'
+                }
+                return Response(message, status=status.HTTP_400_BAD_REQUEST)
+
+        for items in packages or []:
+
+            res_data = {}
+            purl = items.get('purl')
+
+            res_data['purl'] = purl
+            content_type = items.get('content_type')
```

Comment:
I just thought about it while I was looking at this: instead of passing integer choices for `content_type`, I think it would be more descriptive to use the text labels for those choices (https://github.com/nexB/purldb/blob/main/packagedb/models.py#L457) 

A request would look like this:
```
        {
          "purls": [
            {"purl": "pkg:npm/less@1.0.32", "content_type": "curation"}
          ],
          "uuid" : "b67ceb49-1538-481f-a572-431062f382gg"
        }
```

Reply:
Have changed it